lib/model.js

```diff
@@ -2676,6 +2680,9 @@ Model.findByIdAndUpdate = function(id, update, options, callback) {
       throw new TypeError(msg);
     }
     return this.findOneAndUpdate({ _id: id }, undefined);
+  } else if (id === undefined) {
```

Comment:
Our tests rely on `Model.findByIdAndUpdate()` with no arguments working, so can we make this `else if (arguments.length > 0 && id === undefined)`?

Reply:
Thank you for bringing this up. I forgot to consider that. I went ahead and added this to the condition